server/listener.go

```diff
@@ -76,6 +77,12 @@ func (l *Listener) Accept() {
 			continue
 		}
 
+		// Configure read timeouts on this connection
+		// TODO: use timeouts from the live server values
+		if l.cfg.ConnReadTimeout != 0 || l.cfg.ConnWriteTimeout != 0 {
```

Comment:
if ConnReadTimeout != 0 and ConnWriteTimeout == 0, is that a problem?  does netutil.NewConnWithTimeouts handle 0s?  If so why is the if needed at all?

Reply:
This is a good question. Just ripped this code from vitess where the logic is identical